packages/lesswrong/server/repos/UltraFeedEventsRepo.ts

```diff
@@ -0,0 +1,85 @@
+import crypto from 'crypto';
+import AbstractRepo from './AbstractRepo';
+import UltraFeedEvents from '../collections/ultraFeedEvents/collection';
+import groupBy from 'lodash/groupBy';
+import { recordPerfMetrics } from './perfMetricWrapper';
+
+class UltraFeedEventsRepo extends AbstractRepo<'UltraFeedEvents'> {
+  constructor() {
+    super(UltraFeedEvents);
+  }
+
+  /**
+   * Generates a stable hash ID for a comment thread based on its comment IDs (sensitive to sort order).
+   * This MUST match the hash generation logic used in the resolver when checking against served threads.
+   */
+  public static generateThreadHash(commentIds: string[]): string {
```

Comment:
This doesn't need to live on the repo (and probably shouldn't, since it doesn't have anything to do with interacting the the database).

Reply:
Oh I see, you're also using it in `getRecentlyServedCommentThreadHashes`.  Hm...

Reply:
yeah, I was thinking that

Reply:
moved it anyway